processing/src/main/java/io/druid/query/aggregation/JavaScriptAggregatorFactory.java

```diff
@@ -339,11 +340,11 @@ public double aggregate(final double current, final BaseObjectColumnValueSelecto
             if (arg != null && arg.getClass().isArray()) {
               // Context.javaToJS on an array sort of works, although it returns false for Array.isArray(...) and
               // may have other issues too. Let's just copy the array and wrap that.
-              final Object[] arrayAsObjectArray = new Object[Array.getLength(arg)];
-              for (int j = 0; j < Array.getLength(arg); j++) {
-                arrayAsObjectArray[j] = Array.get(arg, j);
-              }
-              args[i + 1] = cx.newArray(scope, arrayAsObjectArray);
+              args[i + 1] = cx.newArray(scope, arrayToObjectArray(arg));
+            } else if (arg instanceof List) {
+              // Using toArray(Object[]), instead of just toArray(), because Arrays.asList()'s impl and similar List
+              // impls could clone the underlying array in toArray(), that could be not Object[], but e. g. String[].
+              args[i + 1] = cx.newArray(scope, ((List) arg).toArray(ObjectArrays.EMPTY_ARRAY));
```

Comment:
do we have a test case for this ?

Reply:
This change is going to be required in the following PR, where I replace some uses of Java arrays with lists. It's going to be tested then.